### PR TITLE
Refine `view_dependencies` query to return user-conrolled objects

### LINKED ISSUE #18: Bug in 1.3.1 if a dependency is `user_scheduler_jobs`
If I have an mview that has a dependency on `user_scheduler_jobs` then schema dumping fails with a KeyError. This only occurs in 1.3.1, so I'm guessing (but haven't yet proved) that it occurs [here](https://github.com/cdinger/scenic-oracle_adapter/compare/v1.3.0...v1.3.1#diff-702381f54d28ba75a5ae074024795ae0362356a59e454ec5ed7d4df1140857d6R86) as that's the only part of the new code that does anything with hashes.

I'm going to dig in to this more.

### PR DESCRIPTION
**Pull request by whit0694**
Closes #18

As mentioned in https://github.com/cdinger/scenic-oracle_adapter/issues/18#issuecomment-2206657472 this updates the query so that it only returns a dependency if it exists in `user_objects`. So dependencies such as `user_scheduler_jobs` or other Oracle-controlled objects are excluded.

### CODE REVIEW

File: lib/scenic/adapters/oracle.rb [hunk 1]
```diff
@@ -76,6 +76,7 @@ def view_dependencies
           from user_objects uo
             left join user_dependencies ud on uo.object_name = ud.name
               and ud.referenced_type in ('VIEW', 'MATERIALIZED VIEW')
+              and ud.object_name in (select object_name from user_objects)
```
Comment (cdinger) [inline lib/scenic/adapters/oracle.rb]: I think you meant `ud.referenced_name` here:
```suggestion
              and ud.referenced_name in (select object_name from user_objects)
```

I've also added a spec to exercise this in main. Could you rebase that into this branch so we can verify that it fixes the issue?

**Comment by whit0694** [comment]
This should be done. Just waiting for tests to run, then I'll rebase. 

https://github.com/cdinger/scenic-oracle_adapter/pull/19/commits/c75af155431604a53a87eef8e708c1e6701d276f sneaks in one more minor tweak. Like I said, it's a weird edge case. But it's definitely one we have!

**Comment by whit0694** [comment]
Tests pass.

**Comment by cdinger** [comment]
Thanks again, @whit0694!
